src/ScrapeRepos.ts

```diff
@@ -0,0 +1,428 @@
+        developers.push({ name, username })
+      })
+
+      // get the repo name
+      htmlC('h1.h4.lh-condensed a').each((i: number, el: cheerio.Element) => {
+        const repo: string = htmlC(el).attr('href')?.substr(1) ?? ''
+        // check if the repo exists
+        if (repo) {
+          const split = repo.split('/')
+          developerRepos.push({ username: split[0], repo: split[1] })
+        }
+      })
+
+      // correctly merge the two arrays
+      const repo = ''
+      return developers.map((developer) => {
+        const matchingRepo = developerRepos.find((repo) => repo.username === developer.username)
+        return { ...developer, ...(matchingRepo || { repo }) }
+      })
+    })
+    .catch((error) => {
+      throw new Error("Couldn't get trending developers")
+    })
+}
+
+/** This method retrieves the stargazers from a page of a GitHub repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github Access token
+ * @param {number} page - Page Number of the stargazers to retrieve
+ * @returns Promise<Object>: A promise that resolves to the stargazers of the repository
+ */
+async function getRepoStargazers(
+  repo: string,
+  token?: string,
+  page?: number
+): Promise<AxiosResponse<StargazersData[]>> {
+  let url = `https://api.github.com/repos/${repo}/stargazers?per_page=${DEFAULT_PER_PAGE}`
+
+  if (page !== undefined) {
+    url = `${url}&page=${page}`
+  }
+  return axios.get(url, {
+    headers: {
+      Accept: 'application/vnd.github.v3.star+json',
+      Authorization: token ? `token ${token}` : ''
+    }
+  })
+}
+
+/** Retrieves the total count of stargazers for a Github repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github access token
+ * @returns Promise<number> : A promise that resolves to the total count of stargzers for the repository
+ */
+async function getRepoStargazersCount(repo: string, token?: string): Promise<number> {
+  const response: AxiosResponse<StargazerCount> = await axios.get(
+    `https://api.github.com/repos/${repo}`,
+    {
+      headers: {
+        Accept: 'application/vnd.github.v3.star+json',
+        Authorization: token ? `token ${token}` : ''
+      }
+    }
+  )
+
+  const data: StargazerCount = response.data
+  return data.stargazers_count
+}
+
+/** Retrieves the star records (star count by date) of a Github repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github Access token
+ * @param {number} maxRequestAmount - Maximum number of API requests to make to retrieve the stars
+ * The higher this value is the more accurate is going to be the graph of the star history
+ * @returns Promise<Array<{ date: string, count: number }>>: A promise that resolves to an array of star records
+ */
+async function getRepoStarRecords(repo: string, token: string, maxRequestAmount: number) {
```

Comment:
Please put some more comments into the code that explain the parts of this function. Using variable names like `patchRes`, `regResult` does not make it easier to understand it. 
Or is this code from star-history?

Reply:
This code is 100% from star history broken down into multiple methods because of ESLint

Reply:
Allright.